ivy/functional/ivy/data_type.py

```diff
@@ -510,7 +510,9 @@ def can_cast(
         b: true
     }
     """
-    return current_backend(from_).can_cast(from_, to)
+    if isinstance(from_, ivy.Dtype):
+        return (from_, to) in ivy.promotion_table
+    return (from_.dtype, to) in ivy.promotion_table
```

Comment:
I think this should be `ivy.as_ivy_dtype(from_.dtype)` instead or otherwise we could just use the `inputs_to_ivy_arrays` decorator for this function instead of the `inputs_to_native_arrays` one. I think the latter is a better option

Reply:
done